### ISSUE TITLE
[Bug]: Accessing Internal Storage from Flatpak Apps

### ISSUE THREAD

**Issue by LuigiWriter** [flatpak/flatpak#4985]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak/issues) for a bug that matches the one I want to file, without success.
- [X] If this is an issue with a particular app, I have tried filing it in the appropriate issue tracker for the app (e.g. under https://github.com/flathub/) and determined that it is an issue with Flatpak itself.
- [X] This issue is not a report of a security vulnerability (see [here](https://github.com/flatpak/flatpak/blob/main/SECURITY.md) if you need to report a security issue).

### Flatpak version

Flatpak 1.12.4

### What Linux distribution are you using?

Debian

### Linux distribution version

5.4.0-121-generic #137~18.04.1-Ubuntu

### What architecture are you using?

x86_64

### How to reproduce

While similar to  "Accessing External storage from Flatpak apps #2713" this is a problem accessing Internal storage, specifically files and directories in / in the system storage.

Specific Distro: Ubuntu 18.04

1. Install GThumb, Dolphin and GTG from flatpak
2. Attempt to access files and folders from them.

- From GThumb, the only directory accessible is ~/Pictures. No other directories in home are shown.
- GTG 0.6 cannot access ~/.var/app/org.gnome.GTG/data/gtg/ insteads shows a static task list.
- Dolphin lists some files and directories but not others which are listed in Nautilus. 
- A Dolphin baloo search returns "file or folder does not exist." for files and folders which are shown in Nautilus

3. Uninstall flatpak versions of GThumb, Dolphin and GTG
4. Install Ubuntu versions of the same.

- GTG 0.3 accesses ~/.local/share/gtg/
- Dolphin lists all files and directories shown in Nautilus
- GThumb accesses all directories in ~/

### Expected Behavior

- GThumb accesses all directories in ~/
- GTG 0.3 accesses ~/.local/share/gtg/
- Dolphin lists all files and directories shown in Nautilus

### Actual Behavior

- From GThumb, the only directory accessible is ~/Pictures. No other directories in home are shown. Also only shows photos that were indexed previously by the Ubuntu Software Shotwell.
- GTG 0.6 cannot access ~/.var/app/org.gnome.GTG/data/gtg/ insteads shows a static task list.
- Dolphin lists some files and directories but not others. In other words a truncated list compared with Nautilus 
- A Dolphin baloo search returns "file or folder does not exist." for files and folders which are shown in Nautilus

### Additional Information

As I discover the same problems with the following apps, I will be with irritation replacing them with the Ubuntu available versions.  I say with irritation because the flatpak versions are usually far more recent than the Ubuntu Software versions. 

**Whether this is unique to Ubuntu 18.04 or not, I cannot say.**
I would welcome comments from users of other distros and Ubuntu 20.04 and 22.04 as to whether they can reproduce this problem as well.

**Comment by Neui** [comment]
Coming from the GTG side here. Flatpak does [sandboxing](https://docs.flatpak.org/en/latest/sandbox-permissions.html) so apps can't access the entirety of your home (and system) folders by default and instead stuff is "punched in" (such as [GThumb allowing to access Pictures folder](https://github.com/flathub/org.gnome.gThumb/blob/master/org.gnome.gThumb.yml#L18); GTG [doesn't need any special directory accesses](https://github.com/flathub/org.gnome.GTG/blob/master/org.gnome.GTG.json#L7-L13)). GTG follows [XDG base dirs](https://wiki.archlinux.org/title/XDG_Base_Directory) and flatpak sets `~/.var/app/org.gnome.GTG/data/` as the data location that GTG follows (to make it easier to separate stuff between applications in a generic way for flatpak).
I am not sure why you think `~/.var/app/org.gnome.GTG/data/gtg/` can't be accessed by GTG v0.6 (I assume the flatpak version). If you're have non-flatpak GTG data, then you unfortunately need to migrate the data yourself currently (basically means copying the `gtg` folders in the `.config/(gtg)` and `.local/share/(gtg)` folders to the flatpaked subdirectories).

**Comment by LuigiWriter** [comment]
Some how I did not get a notice of your comment. 
Thank you very much for the explanation. 
I have moved up to new hardware and Ubuntu Studio 23.04 and need to see how to upgrade to 23.10 which I see is now out.
 GTG v0.6 Flatpak now has no problem accessing ~/.var/app/org.gnome.GTG/data/gtg/
apologies for this thank you being over a year late. Again Thank you for helping me understand the intricacies of packaging and OS's
